src/metabase/data_apps/models.clj

```diff
@@ -109,8 +113,44 @@
 ;;                                       Serialization                                            ;;
 ;;------------------------------------------------------------------------------------------------;;
 
-(comment
-  #_todo)
+(declare released-definition)
+
+(defmethod serdes/make-spec "DataAppRelease" [_model-name _opts]
+  {:copy      [:retracted]
+   :skip      []
+   :transform {:created_at        (serdes/date)
+               :updated_at        (serdes/date)
+               :creator_id        (serdes/fk :model/User)
+               :app_id            (serdes/fk :model/DataApp)
+               :app_definition_id (serdes/parent-ref)}})
+
+(defmethod serdes/make-spec "DataAppDefinition" [_model-name opts]
+  {:copy      [:revision_number :config]
+   :skip      []
+   :transform {:created_at (serdes/date)
+               :creator_id (serdes/fk :model/User)
+               :app_id     (serdes/parent-ref)
+               :release    (serdes/nested :model/DataAppRelease :app_definition_id opts)}})
+
+(defn- export-released-definition
+  "Export the released definition for a data app, if it exists."
+  [data-app]
+  (when-let [definition (released-definition (:id data-app))]
+    [(serdes/extract-one "DataAppDefinition" {} (assoc definition :release [(t2/select-one :model/DataAppRelease :app_definition_id (:id definition))]))]))
```

Comment:
Related to my point on the format, I think it makes more sense to select the latest release, and then select its definition. This also avoids a perverse (and hopefully impossible) edge case where are multiple releases for a definition, and we don't select the newest one.